docs/concepts/cluster_queue.md

```diff
@@ -146,16 +147,25 @@ ClusterQueues can be grouped in _cohorts_. ClusterQueues that belong to the
 same cohort can borrow unused quota from each other.
 
 To add a ClusterQueue to a cohort, specify the name of the cohort in the
-`.spec.cohort` field. All the ClusterQueues that have a matching `spec.cohort`
-are part of the same cohort. If the `spec.cohort` field is empty, the
-ClusterQueue doesn't belong to any cohort, and thus it cannot borrow quota from
-any other ClusterQueue.
+`.spec.cohort` field. All ClusterQueues that have a matching `spec.cohort` are
+part of the same cohort. If the `spec.cohort` field is empty, the ClusterQueue
+doesn't belong to any cohort, and thus it cannot borrow quota from any other
+ClusterQueue.
 
-Borrowing happens per-flavor. If the ClusterQueues belonging to a cohort
-reference different flavors in `.spec.requestableResources[*].flavors[*].resourceFlavors`,
-they won't be able to share quota among them.
+### Flavors and borrowing semantics
 
-### Borrowing semantics example
+When borrowing, Kueue satisfies the following semantics:
+
+- When assigning flavors, Kueue goes through the list of flavors in
+  `.spec.requestableResources[*].flavors`. For each flavor, Kueue attempts to
+  fit the workload using the guaranteed quota of the ClusterQueue or the unused
+  quota in the cohort. If the workload doesn't fit, Kueue proceeds evaluating
+  the next flavor in the list.
+- Borrowing happens per-flavor. If the ClusterQueues belonging to a cohort
+  reference different flavors in `.spec.requestableResources[*].flavors[*].resourceFlavors`,
+  they won't be able to share quota among them.
```

Comment:
The paragraph seems to suggest that a flavor must be defined by all CQs in the cohort for it to be sharable, which is not the case. So perhaps we can rephrase to describe the semantics from a single CQ perspective:

Borrowing happens per-flavor. A ClusterQueue can only borrow quota of flavors it defines.

Reply:
I see that I overly complicated the text :)